src/Analyzer/FileVisitor.php

```diff
@@ -113,6 +113,10 @@ private function addParamDependency(Node\Param $node): void
             return;
         }
 
+        if (method_exists($node->type, 'isSpecialClassName') && ($node->type->isSpecialClassName())) {
```

Comment:
When using `self` or `static` as param type the `$node` object is an instance of `PhpParser\Node\Name`, so this condition could be changed (for example) as follow:
`if (($node->type instanceof Node\Name) && ($node->type->isSpecialClassName())) {`

... but I prefer the actual condition that is more generic because the method `isSpecialClassName()` exists in more classes, for example in `PhpParser\Node\Name` and `PhpParser\Node\Identifier`.

What do you think about ?

Reply:
We can try to use the actual condition and we can understand if we don't have covered well this case ok?